services/MerchStoreService.ts

```diff
@@ -100,47 +100,70 @@ export default class MerchStoreService {
     const item = await this.transactions.readOnly(async (txn) => Repositories
       .merchStoreItem(txn)
       .findByUuid(uuid));
-    if (!item) throw new NotFoundError('Item not found');
+    if (!item) throw new NotFoundError('Merch item not found');
     return item.getPublicMerchItem();
   }
 
   public async createItem(item: MerchItem): Promise<MerchandiseItemModel> {
     return this.transactions.readWrite(async (txn) => {
+      item.hasVariantsEnabled ??= false;
+      this.verifyVariantAndOptionConsistency(item);
+
       const collection = await Repositories.merchStoreCollection(txn).findByUuid(item.collection);
-      if (!collection) throw new NotFoundError('Collection not found');
+      if (!collection) throw new NotFoundError('Merch collection not found');
+
       const merchItemRepository = Repositories.merchStoreItem(txn);
       const merchItem = MerchandiseItemModel.create({ ...item, collection });
       await merchItemRepository.upsertMerchItem(merchItem);
       return merchItemRepository.findByUuid(merchItem.uuid);
     });
   }
 
+  // verify that item.hasVariantsEnabled is consistent with the item options length and option type consistency
+  // as defined by the merch store functionality
+  private verifyVariantAndOptionConsistency(item: MerchItem | MerchandiseItemModel) {
```

Comment:
Where is this defined besides right here in the code? Also this can be static, and can the parameter be only type?

```suggestion
  /**
   * Verify that items have valid options. An item with variants disabled cannot have multiple 
   * options, and an item with variants enabled cannot have multiple option types.
   */
   private static verifyItemHasValidOptions(item: MerchandiseItemModel) {
```

Reply:
I don't believe we can use just 1 parameter since the function is called for both `MerchItem` and `MerchandiseItemModel` objects, unless we define a common interface for the two